components/schema-migrator/run.sh

```diff
@@ -151,6 +151,20 @@ if [[ "${DIRECTION}" == "up" ]]; then
   rm  ${MIGRATION_STORAGE_PATH}/* || true
   cp -R migrations/${MIGRATION_PATH}/. ${MIGRATION_STORAGE_PATH}
 
+  TABLE_EXISTS=$(PGPASSWORD="${DB_PASSWORD}" psql -qtAX -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}" -c "SELECT EXISTS (SELECT FROM pg_tables WHERE schemaname = 'public' AND tablename  = 'schema_migrations');")
+  echo "Table \"schema_migrations\" exists: $TABLE_EXISTS"
+  if [[ "$TABLE_EXISTS" == t ]]; then
+    DATABASE_STATE=$(PGPASSWORD="${DB_PASSWORD}" psql -qtAX -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}" -c "SELECT dirty FROM schema_migrations")
+    echo "DB state is dirty: $DATABASE_STATE"
+    if [[ "$DATABASE_STATE" == t ]]; then
+      CURRENT_VERSION=$(PGPASSWORD="${DB_PASSWORD}" psql -qtAX -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}" -c "SELECT version FROM schema_migrations")
+      echo "Current schema version: $CURRENT_VERSION"
+      LAST_SUCCESSFUL_MIGRATION=$(ls -lr migrations/${MIGRATION_PATH} | grep -i -A 1 ${CURRENT_VERSION} | tail -n 1 | tr -s ' ' | cut -d ' ' -f9 | cut -d '_' -f1)
```

Comment:
what's the benefit of using this approach instead of
```
LAST_SUCCESSFUL_MIGRATION=$(currentVersion)
```

Reply:
Lets say we have migration 11 - a good one and migration 12 - a faulty one. The first time the migrator job tries to apply the migrations it will fail on the second one and in the DB we will have record (version = 12, dirty = true). The current version in that case is the faulty one - 12. We need to forse the schema to the last successfull version - the version before the faulty one - 11. If we set LAST_SUCCESSFUL_MIGRATION=$(currentVersion) this will mean that the last applyed migration is good which is not the case.